src/AccordionToggle.js

```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { elementType } from 'prop-types-extra';
+
+import { createBootstrapComponent } from './ThemeProvider';
+import AccordionContext from './AccordionContext';
+
+class AccordionToggle extends React.Component {
+  static propTypes = {
+    /**
+     * Set a custom element for this component.
+     */
+    as: elementType,
+
+    /** @default 'accordion-toggler' */
+    bsPrefix: PropTypes.string,
+    /**
+     * A key that corresponds to the collapse component that gets triggered
+     * when this has been clicked.
+     */
+    eventKey: PropTypes.string.isRequired,
+
+    /**
+     * A callback function to perform additional operations with this toggler
+     * gets triggered.
+     */
+    onClick: PropTypes.func,
+  };
+
+  static defaultProps = {
+    as: 'button',
+  };
+
+  handleClick = () => {
+    this.accordionContextOnClick(this.props.eventKey);
+    if (this.props.onClick) this.props.onClick(this.props.eventKey);
+  };
+
+  render() {
+    const {
+      as: Component,
+      bsPrefix,
+      children,
+      className,
+      eventKey,
+      ...props
+    } = this.props;
+
+    if (Component === 'button') props.type = 'button';
```

Comment:
I think instead of introspecting `Component` it's better to set the default prop to something like `props => <button type="button {...props}/>` That said I'm not sure Toggle should render anything itself...I think maybe it should just inject props into `children` or have a render function

Reply:
Meaning the Toggle component should just pass onClick and other props to its children, such as this example?
```jsx
<Accordion.Toggle>
  <button>click me</button> 
</Accordion.Toggle>
```
or Should I make my own `<Accordion.Button>` that grabs onClick from the `Accordion` context?

Reply:
I wouldn't add an AccordionButton, the above tho would work. Maybe in the in near future we also provide a `useAccordionToggle` hook to be cool

Reply:
to clarify, should I pass the `onClick` function to its children via context? I'm not sure how I would determine which child gets the function from context though...